lib/async_hooks.js

```diff
@@ -223,15 +223,18 @@ class AsyncResource {
     return this[trigger_async_id_symbol];
   }
 
-  bind(fn, thisArg = this) {
+  bind(fn, thisArg) {
     validateFunction(fn, 'fn');
-    const ret =
-      FunctionPrototypeBind(
-        this.runInAsyncScope,
-        this,
-        fn,
-        thisArg);
-    ObjectDefineProperties(ret, {
+    const runInAsyncScope = FunctionPrototypeBind(
+      this.runInAsyncScope,
+      this,
+      fn);
+    function bound(...args) {
+      return runInAsyncScope(
+        thisArg !== undefined ? thisArg : this,
+        ...args);
```

Comment:
Can you provide more information about the syntax you are looking for?

Reply:
I'm not looking for a specific syntax, I would like to avoid the use of `...args`. As I said in my previous comment, it relies on a globally mutable method:

```js
// user-land
delete Array.prototype[Symbol.iterator];

// core
function bound(...args) {
  return console.log(this, ...args);
}
bound(); // throws
```

As you can see in this example, using the spread operator let unrelated userland code modify the behavior of a Node.js internal, which is something I'd like to avoid.
To avoid this syntax, you could use `Reflect.apply` to call the function, and/or keep the current algorithm when `thisArg` is not nullish.

Reply:
I'm not sure how to add `thisArg` to the `arguments` object without using any of JavaScript. Is there a method cached in Node that I can use to concat them?

Reply:
You could do
```js
function bound(...args) {
  ArrayPrototypeUnshift(args, thisArg !== undefined ? thisArg : this);
  return ReflectApply(runInAsyncScope, null, args);
}
```
If may or may not be more performant to special case the case where `thisArg` is undefined:
```js
let ret;
if (arguments.length === 1) {
  // thisArg is undefined
  let { runInAsyncScope } = this;
  let that = this;
  ret = function (...args) {
    ArrayPrototypeUnshift(args, this);
    return ReflectApply(runInAsyncScope, that, args);
  }
} else {
  ret = 
    FunctionPrototypeBind(
        this.runInAsyncScope,
        this,
        fn,
        thisArg);
}
```

Reply:
Done! Let me know if the new code works, and thanks for the assist, I'm not used to the restrictions of the Node project.